### ISSUE TITLE
Memory Leak

### ISSUE THREAD

**Issue by simlu** [shevek/parallelgzip#4]
There seems to be a memory leak somewhere... Urg... Will investigate.

**Comment by simlu** [comment]
So there wasn't a memory leak after all. However since there are a lot of byte[] arrays created and destroyed in this class, it's very hard to figure out what is going on elsewhere.

Will create a pull request with significant memory usage improvements.

**Comment by simlu** [comment]
_Background:_ I'm using this to stream many terabyte of zipped data from and to gzip in parallel. This change has made a huge difference in memory performance. @shevek

**Comment by shevek** [comment]
Closing in favour of #5  , which has a review comment now. If we can pass the test suite and a reasonable argument of correctness, I very much want this patch.
